src/utils/Snapshot.js

```diff
@@ -148,3 +153,124 @@ export const splitSnapshotAffectedStructs = (transaction, snapshot) => {
     meta.add(snapshot)
   }
 }
+
+/**
+ * @param {Doc} originDoc
+ * @param {Snapshot} snapshot
+ * @return {Doc}
+ */
+export const createDocFromSnapshot = (originDoc, snapshot) => {
```

Comment:
`Item`s depend on each other and must be applied in a very specific order. The function `resumeStructIntegration` handles the correct integration process. If an `Item` depends on another `Item`, the dependency is integrated first.

In multi-user scenarios, the below integration process will likely result in issues. 

Copying `Item`s might be the fastest approach. But I think first we should implement a simple working implementation that just encodes the snapshotted document and then calls `applyUpdate` to make use of the existing struct integration process.

This is how I would lay out the implementation:

* Starting with a snapshot `{ sv, ds }`.
* Split items to ensure that really just the state is copied that we are interested in: `sv.forEach((clock, client) => getItemCleanStart(transaction, ID(client, clock)))` (note: first check if item `ID(client, clock)` exists, otherwise getItemCleanStart might throw an error).
* Create V2 encoder (v2 is probably faster).
* Write all structs from [0 - clock) to encoder.
* Write `ds` to encoder.
* The encoded document is now a document update containing all updates from the state vector
* Apply encoded document to a fresh Y.Doc